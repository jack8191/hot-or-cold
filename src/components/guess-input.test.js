import React from 'react'
import {shallow, mount} from 'enzyme'

import GuessInput from './guess-input'
import toJson from 'enzyme-to-json'

describe('<GuessInput />', () => {
    it('Renders without crashing', () => {
        shallow(<GuessInput />)
    })
    it('Renders an empty form', () => {
        const wrapper = shallow(<GuessInput/>)
        expect(wrapper.is('form'))
    })
    it('Accepts an input when the user submits a form', () => {
        const callback = jest.fn()
        const wrapper = mount(<GuessInput addInput={callback}/>)
        const input = "2"
        wrapper.instance()
        wrapper.update()
        wrapper.find('input[type="number"]').instance().value = input
        wrapper.simulate('submit')
        expect(callback).toHaveBeenCalledWith(input)
        
    })
    it('renders correctly', () => {
        const wrapper = shallow(<GuessInput/>)
       
        expect(toJson(wrapper)).toMatchSnapshot();
      });
})