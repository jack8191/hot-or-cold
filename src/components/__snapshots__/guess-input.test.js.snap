// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuessInput /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoComplete="off"
    className="text"
    id="userGuess"
    max="101"
    min="1"
    name="userGuess"
    type="number"
  />
  <button
    className="button"
    id="guessButton"
    name="submit"
    type="submit"
  >
    Guess
  </button>
</form>
`;
